src/app/features/test-records/components/test-type-select/test-type-select.component.ts

```diff
@@ -8,12 +12,23 @@ import { Observable } from 'rxjs';
   templateUrl: './test-type-select.component.html',
   styleUrls: ['./test-type-select.component.scss'],
 })
-export class TestTypeSelectComponent {
+export class TestTypeSelectComponent implements OnInit {
   @Output() testTypeSelected = new EventEmitter<TestType>();
 
   categories: Array<TestTypeCategory> = [];
 
-  constructor(private testTypesService: TestTypesService) {}
+  constructor(
+    private testTypesService: TestTypesService,
+    private technicalRecordService: TechnicalRecordService,
+    private router: Router,
+  ) {}
+  ngOnInit(): void {
+    this.technicalRecordService.techRecordHistory$.subscribe((recordHistory) => {
+      if (!recordHistory) {
+        void this.router.navigate(['tech-records/:systemNumber/:createdTimestamp']);
```

Comment:
The systemNumber or createdTimestamp needs to be added

Reply:
Nice shout-out. I will add it now.